pipeline/ffmpeg.go

```diff
@@ -135,6 +135,21 @@ func (f *ffmpeg) HandleStartUploadJob(job *JobInfo) (*HandlerOutput, error) {
 
 	job.sourceSegments = len(sourceManifest.GetAllSegments())
 
+	sourceSegmentURLs, err := transcode.GetSourceSegmentURLs(transcodeRequest.SourceManifestURL, sourceManifest)
+	if err != nil {
+		return nil, fmt.Errorf("error fetching source segment URLs: %w", err)
+	}
+	for _, seg := range sourceSegmentURLs {
```

Comment:
Do we need to probe all of them? Could we maybe do the first and last or something?